src/internal/operations/pathtransformer/restore_path_transformer.go

```diff
@@ -0,0 +1,180 @@
+	// was in the root of the data type.
+	elems := repoRef.Folders()
+
+	if ent.OneDrive != nil || ent.SharePoint != nil {
+		dp, err := path.ToDrivePath(repoRef)
+		if err != nil {
+			return nil, clues.Wrap(err, "fallback for LocationRef")
+		}
+
+		elems = append([]string{dp.Root}, dp.Folders...)
+	}
+
+	return path.Builder{}.Append(elems...), nil
+}
+
+func basicPathMerge(repoRef path.Path, locRef *path.Builder) (path.Path, error) {
+	if len(locRef.Elements()) == 0 {
+		res, err := path.ServicePrefix(
+			repoRef.Tenant(),
+			repoRef.ResourceOwner(),
+			repoRef.Service(),
+			repoRef.Category())
+		if err != nil {
+			return nil, clues.Wrap(err, "getting prefix for empty location")
+		}
+
+		return res, nil
+	}
+
+	return locRef.ToDataLayerPath(
+		repoRef.Tenant(),
+		repoRef.ResourceOwner(),
+		repoRef.Service(),
+		repoRef.Category(),
+		false)
+}
+
+func drivePathMerge(
+	ent *details.Entry,
+	repoRef path.Path,
+	locRef *path.Builder,
+) (path.Path, error) {
+	// Try getting the drive ID from the item. Not all details versions had it
+	// though.
+	var driveID string
+
+	if ent.SharePoint != nil {
+		driveID = ent.SharePoint.DriveID
+	} else if ent.OneDrive != nil {
+		driveID = ent.OneDrive.DriveID
+	}
+
+	// Fallback to trying to get from RepoRef.
+	if len(driveID) == 0 {
+		odp, err := path.ToDrivePath(repoRef)
+		if err != nil {
+			return nil, clues.Wrap(err, "fallback getting DriveID")
+		}
+
+		driveID = odp.DriveID
+	}
+
+	return basicPathMerge(
+		repoRef,
+		path.FormatDriveFolders(driveID, locRef.Elements()...))
+}
+
+func restorePathsForItem(
+	ctx context.Context,
+	backupVersion int,
+	ent *details.Entry,
+) (path.RestorePaths, error) {
+	res := path.RestorePaths{}
+
+	repoRef, err := path.FromDataLayerPath(ent.RepoRef, true)
+	if err != nil {
+		err = clues.Wrap(err, "parsing details path after reduction").
+			WithClues(ctx).
+			With("storage_path", clues.Hide(ent.RepoRef), "location", clues.Hide(ent.LocationRef))
+
+		return res, err
+	}
+
+	res.StoragePath = repoRef
+	ctx = clues.Add(ctx, "storage_path", repoRef)
+
+	// Get the LocationRef so we can munge it onto our path.
+	locRef, err := locationRef(ent, repoRef, backupVersion)
+	if err != nil {
+		err = clues.Wrap(err, "parsing LocationRef after reduction").
+			WithClues(ctx).
+			With("location", clues.Hide(ent.LocationRef))
+
+		return res, err
+	}
+
+	ctx = clues.Add(ctx, "location", locRef)
+
+	// Now figure out what type of ent it is and munge the path accordingly.
+	// Eventually we're going to need munging for:
+	//   * Exchange Calendars (different folder handling)
+	//   * Exchange Email/Contacts
+	//   * OneDrive/SharePoint (needs drive information)
+	if ent.Exchange != nil {
+		// TODO(ashmrtn): Eventually make Events have it's own function to handle
+		// setting the restore destination properly.
+		res.RestorePath, err = basicPathMerge(repoRef, locRef)
+	} else if ent.OneDrive != nil || ent.SharePoint != nil {
```

Comment:
Not all sharepoint paths are drive paths.  While we aren't supporting behavior for pages and lists live, we do have tests in place for those categories (not 100% clear on their e2e integration at this moment).  Might be safest to include category comparison in this decision as well.